### DISCUSSION TITLE
Add token to the HLS

### DISCUSSION THREAD

**Discussion by thanhlnbka** [bluenviron/mediamtx#1647] [Questions]
How do I add a token to the hls extension so that I can avoid having this hls link copied to another browser and viewable? 
For example, add a token like this https://localhost:8888/mystream/8b7a950d1776c13b32def2426db14b13 and this token only allows 1 viewable place, then when I reload the page, it will recreate a new token. 
The purpose is to avoid others copying the link on the browser and taking it somewhere else to view it.
Thanks for reading !

**Comment by volkankoc** [comment]
I want this too

**Comment by volkankoc** [comment]
please help

**Comment by thanhlnbka** [comment]
@volkankoc I interfered with webrtc's websocket and added a piece of code to verify the token.
I have built a flask api to allow token verification. If the token has already been used, it will not be used a second time and the server side websocket will not be connected.
Here is my repo to test this case, hope it helps you https://github.com/thanhlnbka/TestMediamtx/tree/main

**Reply by volkankoc** [reply]
> @volkankoc I interfered with webrtc's websocket and added a piece of code to verify the token. I have built a flask api to allow token verification. If the token has already been used, it will not be used a second time and the server side websocket will not be connected. Here is my repo to test this case, hope it helps you https://github.com/thanhlnbka/TestMediamtx/tree/main

Thank you.
But,  how to will use this

**Comment by aler9** [comment]
Hello, this is not a feature implemented by server. Access control is implemented in a myriad of ways, therefore adding one-click access control into the server would not cover most cases, like for example, cookie-based access control, time-based access control or token-based access control and so on.

Nonetheless, the advantage of HLS and WebRTC is that they are based on HTTP, therefore you can build any custom logic you want by placing a reverse proxy in front of the server, written in a language of your choice. Personally i seldom use the server in a standalone way, but almost always behind a more complex infrastructure that provides a custom frontend and multiple layers of backend.

**Comment by volkankoc** [comment]
Can you tell us how to configure this? I did not fully understand.

[a comment by an automated account is not shown]
